open-craft/xblock-skytap
# Persist preferred keyboard layout

cf. [OC-2504](https://tasks.opencraft.com/browse/OC-2504)

**Test instructions**

0. Add keyboard layouts to `XBLOCK_SETTINGS` in `lms.env.json`:

    ```json
    "XBLOCK_SETTINGS": {
        "skytap": {
            "keyboard_layouts": {
                "nl-be": "Dutch-Belgium",
                "uk": "English (UK)",
                "us": "English (US)",
                "fi": "Finnish",
                "fr": "French",
                "fr-be": "French-Belgium",
                "fr-ch": "French-Switzerland",
                "de": "German",
                "de-ch": "German-Switzerland",
                "is": "Icelandic",
                "it": "Italian",
                "jp": "Japanese",
                "no": "Norwegian",
                "pt": "Polish",
                "es": "Spanish"
            }
        }
    },
    ```

1. Add Skytap XBlock to a unit.

2. Access the unit in the LMS.
    * Observe that the default choice for the keyboard layout option is "English (US)".
    * Observe that the list of choices is sorted alphabetically.

3. Select a different option, then click "Launch".

4. Refresh the page. Observe that the keyboard layout option now defaults to the keyboard layout option you selected in the previous step.

**Reviewers**
